components/playlist/browser/playlist_media_file_downloader.h

```diff
@@ -131,6 +132,10 @@ class PlaylistMediaFileDownloader
       download_item_observation_{this};
 
   // All below variables are only for playlist creation.
+  // TODO(sko) These are really fragile and unreliable. DownloadManager invokes
+  // many async callbacks, at the timing of the callback, these variables might
+  // be out of sync. We should wrap these into a class that matches the actual
+  // download item 1:1.
```

Comment:
Exactly. Even though massively disadvantageous, this "transient states as member variables" style is quite prevalent in Playlist for some reason (e.g. `PlaylistDownloadRequestManager`, `PlaylistMediaFileDownloadManager`, etc.). I'm not even sure why we have to queue up download requests, as you can start watching a video while it's downloading? But then even if we want to, there are other ways. Anyway, approving now, but also adding these to my "classes to revisit" list. 🙂

Reply:
Yeah, I'm not sure it's worth queueing.